woocommerce/woocommerce-android
# Feature/8635 onboarding tracking part ii

Part II: #8635 


### Description
This PR adds tracking for `store_onboarding_task_completed` (only for native onboarding task, webviews are not tracked with this event for now)

![Screenshot 2023-03-29 at 10 26 45](https://user-images.githubusercontent.com/2663464/228473674-90eca73c-5033-4381-9d5c-c2f63e448915.png)

Many files changed, but don't panic all changes are small and simple. 

Essentially, the approach was to add a `source` parameter to those onboarding tasks that can be triggered from different places on the app, to ensure that we only track `store_onboarding_task_completed` when the flow was started from the onboarding list. 

**Edit:** Adittionaly to the `store_onboarding_task_completed` I've added tracking for `FREE_TRIAL_UPGRADE_NOW_TAPPED` triggered when clicking on the free trial banner inside launch store task. 

### Testing instructions

1. Complete add product task and check logcat for the following log: `store_onboarding_task_completed, Properties: {"task":"products","blog_id":217026331,"is_wpcom_store":true,"is_debug":true}`
2. Check that adding product from products tab doesn't fire that event. 
3. Complete launch your store and check logcat for the following log: `Tracked: store_onboarding_task_completed, Properties: {"task":"launch_site","blog_id":217026331,"is_wpcom_store":true,"is_debug":true}`
4. Complete customize domain and check logcat for the following log: `Tracked: store_onboarding_task_completed, Properties: {"task":"add_domain","blog_id":217026331,"is_wpcom_store":true,"is_debug":true}`. Note that the backend takes some time to mark this task as completed. So you might still see the task available in the onboarding list for a bit. 
5. Check that adding a domain from settings doesn't fire that event.

**Free trial banner test**
1. Open launch store task from a free trial site
2. Click on the banner and check the log for tracked event: `Tracked: free_trial_upgrade_now_tapped, Properties: {"source":"banner","blog_id":217275144,"is_wpcom_store":true,"is_debug":true}`
